File: C/프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(int a, int b) {
    long long answer = 0;
    answer = ((long)abs(a-b)+1)*(long)(a+b)/2;
    
    return answer;
}

File: Kotlin/프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.kt
class Solution {
    fun solution(a: Int, b: Int): Long {
        var answer: Long = 0
        
        when {
            (a < b) -> for (i in a..b) answer += i
            (a > b) -> for (i in b..a) answer += i
            else -> return a.toLong()
        }
        
        return answer
    }
}